src/main.rs

```diff
@@ -573,9 +588,36 @@ async fn start_ldk() {
 		Arc::clone(&logger),
 	));
 
+	// Step 5: Initialize Persistence
+	let fs_store = Arc::new(FilesystemStore::new(ldk_data_dir.clone().into()));
+	let persister = Arc::new(MonitorUpdatingPersister::new(
+		Arc::clone(&fs_store),
+		Arc::clone(&logger),
+		1000,
+		Arc::clone(&keys_manager),
+		Arc::clone(&keys_manager),
+	));
+	// Alternatively, you can use the `FilesystemStore` as a `Persist` directly, at the cost of
+	// larger `ChannelMonitor` update writes (but no deletion or cleanup):
+	//let persister = Arc::clone(&fs_store);
+
+	// Step 6: Initialize the ChainMonitor
+	let chain_monitor: Arc<ChainMonitor> = Arc::new(chainmonitor::ChainMonitor::new(
+		None,
+		Arc::clone(&broadcaster),
+		Arc::clone(&logger),
+		Arc::clone(&fee_estimator),
+		Arc::clone(&persister),
+	));
+
 	// Step 7: Read ChannelMonitor state from disk
-	let mut channelmonitors =
-		persister.read_channelmonitors(keys_manager.clone(), keys_manager.clone()).unwrap();
+	let mut channelmonitors = persister.read_all_channel_monitors_with_updates(
+		Arc::clone(&bitcoind_client),
+		Arc::clone(&bitcoind_client),
```

Comment:
```suggestion
		&bitcoind_client,
		&bitcoind_client,
```

Change needed to build correctly